go/vt/vtgate/engine/fake_vcursor_test.go

```diff
@@ -397,27 +445,29 @@ func (f *loggingVCursor) ResolveDestinations(keyspace string, ids []*querypb.Val
 
 		switch d := destination.(type) {
 		case key.DestinationAllShards:
-			shards = f.shards
+			for i := 0; i <= len(f.shards); i++ {
+				shards = append(shards, f.makeAShard(i))
+			}
```

Comment:
is it possible for your test to change without changing this part?

Reply:
The original test hardcoded a KeyRange, whereas CFC test needs to dynamically resolve a range in different tests. I couldn't figure out how a single keyrange would work. Do you have suggestions ?

Reply:
I am more concerned about assigning shards, you can change the logic fro resolving of keyrange.
In each test you can assign new shard ranges.